### PR TITLE
Document and fix layouts

### PR DESCRIPTION
**Pull request by josepjaume**
## 🚪 Why?

Layouts are a core part of our application and yet, we haven't put much thought into it.

## 🔑 What?

This adds some proper examples to layouts, and standardizes, well... the StandardLayout. While migrating to Factorial One, I've found out that most layouts either use a full-width content approach (tables?), a narrow content approach (for readability) or fullscreen (org chart).

We're in the process to adding more layouts, but this is a necessary step for backwards compatibility.

### CODE REVIEW

File: lib/experimental/Layouts/InfoPaneLayout/index.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ export const InfoPaneLayout = forwardRef<HTMLDivElement, InfoPaneLayoutProps>(
     return (
       <div
         ref={ref}
-        className="flex h-full flex-col-reverse gap-4 text-f1-foreground sm:flex-row sm:gap-0"
+        className="flex h-full flex-col-reverse gap-4 overflow-auto text-f1-foreground sm:flex-row sm:gap-0"
       >
         <div className="sm:my-6 sm:ms-6 sm:basis-3/4">{mainContent}</div>
```
Comment (sauldom102) [inline lib/experimental/Layouts/InfoPaneLayout/index.tsx:16]: What about passing the `mainContent` as the `children` directly? 👀

**Review by sauldom102 — APPROVED** [review]

File: lib/experimental/Layouts/InfoPaneLayout/index.tsx (hunk 1 above)
Comment (josepjaume) [reply lib/experimental/Layouts/InfoPaneLayout/index.tsx:16]: Absolutely, that should have been done like this.

Comment (josepjaume) [reply lib/experimental/Layouts/InfoPaneLayout/index.tsx:16]: I'll do it!

**Comment by nlopin** [comment]
Are they planned to be used only inside the `ApplicationFrame`?

**Comment by josepjaume** [comment]
Yes @nlopin - I would actually call them Page Layouts. Should i change the name?

File: lib/experimental/Layouts/FullscreenLayout/index.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from "react"
+
+export interface StandardLayoutProps {
+  children: React.ReactNode
+}
+
+export function FullscreenLayout({ children }: StandardLayoutProps) {
+  return <div className="relative flex flex-1 [&>div]:flex-1">{children}</div>
```
Comment (nlopin) [inline lib/experimental/Layouts/FullscreenLayout/index.tsx:8]: What is your thought on using `flex` as a default container across layouts? 

I thought we can have it HTML default value to keep things familiar

File: lib/experimental/Layouts/StandardLayout/index.tsx [hunk 3]
```diff
@@ -1,13 +1,27 @@
...
-export interface StandardLayoutProps {
-  children: React.ReactNode
+export interface StandardLayoutProps
+  extends VariantProps<typeof layoutVariants> {
+  children?: React.ReactNode
 }
 
-export function StandardLayout({ children }: StandardLayoutProps) {
+const layoutVariants = cva(
+  "relative flex w-full flex-1 flex-col gap-4 place-self-center overflow-y-auto px-6 py-5",
```
Comment (nlopin) [inline lib/experimental/Layouts/StandardLayout/index.tsx:11]: why do we set `relative`?

**Comment by nlopin** [comment]
I think every layout must have [`isolate` class](https://tailwindcss.com/docs/isolation) to reduce the possibility of z-index collision with the rest of the frame.

This means all children of the layout will be _below_ the sidebar. It can be an issue itself if we don't use portals for our modals and popups, however I believe we do use them

**Comment by nlopin** [comment]
> Yes @nlopin - I would actually call them Page Layouts. Should i change the name?

no, all good. I tried to understand why do we rely so hard on flex positioning

File: lib/experimental/Layouts/StandardLayout/index.tsx (hunk 3 above)
Comment (josepjaume) [reply lib/experimental/Layouts/StandardLayout/index.tsx]: To kind of create a "box" nobody can escape, even when using `absolute` positioning. Isn't this correct?

File: lib/experimental/Layouts/FullscreenLayout/index.tsx (hunk 2 above)
Comment (josepjaume) [reply lib/experimental/Layouts/FullscreenLayout/index.tsx]: In this case, what I'm trying to do is for the content to stretch. Think the `Org chart` layout, for example. Maybe it's too much and we should just let people create their own layouts in this scenario?

**Comment by josepjaume** [comment]
@nlopin I will do it! Actually, the `isolate` property might be better set at the `page` level to guarantee every layout will always have this.

BTW another interesting thing we need to standardize are loading states. I guess we should create a `PageLayoutBaseProps` interface that every layout extends from.

**Review by nlopin — APPROVED** [review]

**Pull request merged**